homeassistant/components/huawei_solar/config_flow.py

```diff
@@ -0,0 +1,125 @@
+"""Config flow for Huawei Solar integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from huawei_solar import (
+    AsyncHuaweiSolar,
+    ConnectionException,
+    ReadException,
+    register_names as rn,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_SLAVE_IDS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 502
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Required(CONF_SLAVE_IDS, default="0"): str,
```

Comment:
maybe move 0 to a constant a well?

Reply:
Is it possible to have validation of user input for this field? I tried various configurations of voluptuous, but nothing seems to work.

Reply:
What are the possibilities? Would a select work better perhaps, or what would be the validation that you need?

Reply:
It is a comma-separated list of integers.

I would like to validate `[\d]+(, [\d]+)*`

(cfr lines 99-101 where I do the conversion)

Reply:
I fixed it for now by catching the `ValueError` when the conversion into slave-id's fails. It shows the user an informative error.